cmake/coverage.cmake

```
@@ -34,16 +34,31 @@
 find_program(LCOV_PATH lcov)
 find_program(GENHTML_PATH genhtml)
 
+message(STATUS "Building for code coverage")
+
 # add code coverage build type
+if (("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang") OR ("${CMAKE_CXX_COMPILER_ID}" MATCHES "AppleClang"))
```

```suggestion
if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
```

"Clang" MATCHES "Clang" and "AppleClang".